### ISSUE TITLE
Uppercase tranform is not supported in application gateway terraform

### ISSUE THREAD

**Issue by Patrick1411** [hashicorp/terraform-provider-azurerm#27826]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.0

### AzureRM Provider Version

4.7.0

### Affected Resource(s)/Data Source(s)

azurerm_web_application_firewall_policy

### Terraform Configuration Files

```hcl
{
      name             = "BlockExternalAccessToInternalResource"
      action           = "Block"
      enabled          = false
      priority         = 95
      match_conditions = [
        {
          operator        = "Contains"
          transforms      = ["Uppercase"]
          match_values    = ["INT_NHSUN_SAP"]
          match_variables = [
            {
              variable_name = "RequestUri"
            }
          ]
        }
      ]
    },
```

### Debug Output/Panic Output

```shell
Error: expected custom_rules.29.match_conditions.0.transforms.0 to be one of ["HtmlEntityDecode" "Lowercase" "RemoveNulls" "Trim" "UrlDecode" "UrlEncode"], got Uppercase
```

### Expected Behaviour

Uppercase should be allowed

### Actual Behaviour

Uppercase is not allowed during application gateway waf creation

### Steps to Reproduce

terraform init
terraform plan
terraform apply

**Comment by sinbai** [comment]
Hi @Patrick1411 thanks for raising this issue. I have submitted [PR ](https://github.com/hashicorp/terraform-provider-azurerm/pull/27833)to support `Uppercase` for `transforms` property. Could you please track it for more updates?

**Comment by Patrick1411** [comment]
Hi @sinbai thanks for replying me about the issue, so could I know when new version is updated ?

**Comment by rcskosir** [comment]
PR #27833 was merged so it should be going out with v4.8.0 of the provider.
